glidertools/utils.py

```diff
@@ -129,7 +130,15 @@ def merge_dimensions(df1, df2, interp_lim=3):
 
     if same_type:
         df = df1.join(df2, sort=True, how="outer", rsuffix="_drop")
+        # Temporarily remove columns that are dtype datetime64 as these break the interpolation
+        df_no_interpolate = pd.DataFrame()
+        for key in list(df):
+            if pd.api.types.is_datetime64_dtype(df[key]):
+                df_no_interpolate[key] = df[key]
+                df = df.drop(key, axis=1)
         df = df.interpolate(limit=interp_lim).bfill(limit=interp_lim)
+        # Return removed datetime64 columns to dataframe
+        df = df.join(df_no_interpolate)
```

Comment:
I am not quite sure what is actually interpolated, and how that affects the `join` later. Was there a test for this functionality? If not would it be possible to add one? That might help me understand this a bit more clearly.

Reply:
Yeah I'm not clear what's meant to happen here either tbh. Just tried to fix the pandas >1.3.0 error with minimum code changes @tjryankeogh do you know what this interpolation is meant to do?

I can try and write a test to at least check this PR doesn't change the output of the function

Reply:
@callumrollo I think this function was to deal with different sampling resolutions of the CTD and other sensors, so this was an attempt to interpolate them onto a single time dimension.

Reply:
But in that case we need to be more careful here, right? Since the time is basically the interpolant, it cannot just be dropped and readded?